### PR TITLE
Merge the chemical model stuffs to the workflow for v15 runs

### PR DESCRIPTION
**Pull request by zhanglikate**
Some folders have added into the workflow other than the merging udpates:
1)The gfs_post has been added
2)The prepchem_NC has been added
3)The gsi.fd_chem has been  added

### CODE REVIEW

File: scripts/exglobal_fcst_nemsfv3gfs.sh [hunk 1]
```diff
@@ -315,7 +335,15 @@ EOF
         res_latlon_dynamics="fv_increment.nc"
       fi
     fi
-  
+#lzhang: link sfc data from gfs initial conditions
+  for file in $memdir/INPUT/*.nc; do
+    file2=$(echo $(basename $file))
+    fsuf=$(echo $file2 | cut -c1-3)
+    if [ $fsuf = $fsuf = "sfc" ]; then
```
Comment (jkhender) [inline scripts/exglobal_fcst_nemsfv3gfs.sh]: Should line 342 be if [ $fsuf = "sfc" ]; then

File: parm/config/config.base.emc.dyn [hunk 2]
```diff
@@ -49,7 +49,8 @@ export BASE_SVN="@BASE_SVN@"
 #### export CCPP_SUITE="FV3_GSD_v0"        # GSDsuite 
 #### export CCPP_SUITE="FV3_GSD_noah"      # GSDsuite + NOAH LSM  
 #### export CCPP_SUITE="FV3_GFS_v16beta"   # EMC v16beta
-export CCPP_SUITE="FV3_GFS_v15"            # EMC v15
+#export CCPP_SUITE="FV3_GFS_v15"            # EMC v15
+export CCPP_SUITE="GFS_2017_gfdlmp_gsd_chem"            # GSL:GFS_2017_gfdlmp_gsd_chem
```
Comment (jkhender) [inline parm/config/config.base.emc.dyn]: in build_fv3_ccpp_chem.sh, you are compiling only with FV3_GFS_v15_gsd_chem. This needs to be one of the suites you compiled with.  You can list multiple suites when you compile.

Also, the name needs the FV3 at the beginning, "FV3_GFS_2017_gfdlmp_gsd_chem"

File: sorc/build_fv3_ccpp_chem.sh [hunk 3]
```diff
@@ -21,5 +21,10 @@ if [ $target = hera ]; then target=hera.intel ; fi
 cd fv3gfs.fd/
 cd tests/
 #JKH./compile.sh "$target" "CCPP=Y 32BIT=Y STATIC=Y SUITES=FV3_GFS_v15,FV3_GSD_v0,FV3_GSD_noah,FV3_GFS_v16beta" 2 NO NO
-./compile.sh  "$target" "CCPP=Y 32BIT=Y STATIC=Y SUITES=FV3_GFS_v15" 2 NO NO
+#./compile.sh  "$target" "CCPP=Y 32BIT=Y STATIC=Y SUITES=FV3_GFS_v15" 2 NO NO
+#./compile.sh  "$target" "CCPP=Y 32BIT=Y STATIC=Y DEBUG=Y SUITES=FV3_GFS_v15" 2 NO NO  #minimum 
+#./compile.sh "$target" "CCPP=Y 32BIT=Y STATIC=Y SUITES=FV3_GFS_v15" 2 YES YES  #clean everything and recompile
+#./compile.sh "$target" "CCPP=Y 32BIT=Y STATIC=Y SUITES=FV3_GFS_2017_gfdlmp_gsd_chem" 2 YES YES  #clean everything and recompile
+./compile.sh "$target" "CCPP=Y 32BIT=Y STATIC=Y SUITES=FV3_GFS_v15_gsd_chem" 2 YES YES  #clean everything and recompile
```
Comment (jkhender) [inline sorc/build_fv3_ccpp_chem.sh:28]: the suite here should match what is specified in config.base.emc.dyn if you only use 1 to compile with

You can also list multiple suites to compile with

SUITES=FV3_GFS_v15_gsd_chem,FV3_GFS_2017_gfdlmp_gsd_chem"

File: sorc/checkout.sh [hunk 4]
```diff
@@ -20,30 +20,26 @@ echo fv3gfs_ccpp_chem checkout ...
...
     rm -f ${topdir}/checkout-fv3gfs_ccpp_chem.log
     rm fv3gfs.fd
-    git clone --recursive -b gsd/develop-chem https://github.com/haiqinli/ufs-weather-model  fv3gfs_ccpp_chem.fd >> ${topdir}/checkout-fv3gfs_ccpp_chem.log 2>&1
-    cd fv3gfs_ccpp_chem.fd
-    git checkout 433fc66b55e154d7c42fe108f1d8b2ab3f629ec4
-    git submodule sync
-    git submodule update --init --recursive
-    cd ${topdir}
+   git clone --recursive --branch gsd/develop-chem https://github.com/NOAA-GSL/ufs-weather-model  fv3gfs_ccpp_chem.fd >> ${topdir}/checkout-fv3gfs_ccpp_chem.log 2>&1
     ln -fs fv3gfs_ccpp_chem.fd fv3gfs.fd
```
Comment (jkhender) [inline sorc/checkout.sh]: still need lines 24, 26, 27, 28 

    cd fv3gfs_ccpp_chem.fd		
     git submodule sync		
     git submodule update --init --recursive		
     cd ${topdir}

do you want to add the hash as in line 25 so everyone always checks out the same version?

File: sorc/checkout.sh [hunk 5]
```diff
@@ -20,30 +20,26 @@ echo fv3gfs_ccpp_chem checkout ...
...
-    cd fv3gfs_ccpp.fd
-    git checkout 2b5768b19409ba04a37b89833268b7a1d9233139
-    git submodule sync
-    git submodule update --init --recursive
-    cd ${topdir}
-    ln -fs fv3gfs_ccpp.fd fv3gfs.fd
-else
-    echo 'Skip.  Directory fv3gfs_ccpp.fd already exists.'
-fi
+#echo fv3gfs_ccpp checkout ...
```
Comment (jkhender) [inline sorc/checkout.sh]: delete lines 29-42

File: sorc/checkout.sh [hunk 6]
```diff
@@ -79,17 +76,27 @@ else
...
-if [[ ! -d gfs_post.fd ]] ; then
-    rm -f ${topdir}/checkout-gfs_post.log
-    git clone https://github.com/NOAA-EMC/EMC_post.git gfs_post.fd >> ${topdir}/checkout-gfs_post.log 2>&1
-    cd gfs_post.fd
-    git checkout upp_gfsv16_release.v1.0.10
+echo prepchem_NC.fd checkout ...
+if [[ ! -d prepchem_NC.fd ]] ; then
+    rm -f ${topdir}/checkout-prepchem_NC.fd.log
+    git clone  gerrit:GSD-prep-chem prepchem_NC.fd >> ${topdir}/checkout-prepchem_NC.fd.log 2>&1
+    cd prepchem_NC.fd
```
Comment (jkhender) [inline sorc/checkout.sh:73]: line 83 is not needed if you are not checking out a specific hash

**Review by jkhender — CHANGES_REQUESTED** [review]

File: scripts/exglobal_fcst_nemsfv3gfs.sh (hunk 1 above)
Comment (zhanglikate) [reply scripts/exglobal_fcst_nemsfv3gfs.sh]: updated

File: parm/config/config.base.emc.dyn (hunk 2 above)
Comment (zhanglikate) [reply parm/config/config.base.emc.dyn]: updated

File: sorc/build_fv3_ccpp_chem.sh (hunk 3 above)
Comment (zhanglikate) [reply sorc/build_fv3_ccpp_chem.sh:28]: Already updated  the config.based

File: sorc/checkout.sh (hunk 5 above)
Comment (zhanglikate) [reply sorc/checkout.sh]: Updated

File: sorc/checkout.sh (hunk 6 above)
Comment (zhanglikate) [reply sorc/checkout.sh:73]: Which line did you mean?

**Review by zhanglikate — COMMENTED** [review]
Updated based on  the review

File: sorc/checkout.sh (hunk 4 above)
Comment (zhanglikate) [reply sorc/checkout.sh]: updated

File: sorc/checkout.sh (hunk 6 above)
Comment (jkhender) [reply sorc/checkout.sh:73]: line 83 is 
    cd prepchem_NC.fd     (now line 73 in updated file)

**Review by jkhender — CHANGES_REQUESTED** [review]
In checkout.sh,  remove this line

    cd prepchem_NC.fd

**Review by jkhender — APPROVED** [review]

**Pull request merged**